Mathlib/MeasureTheory/Integral/IntervalIntegral.lean

```diff
@@ -373,6 +373,64 @@ theorem Antitone.intervalIntegrable {u : ℝ → E} {a b : ℝ} (hu : Antitone u
 
 end
 
+/-!
+## Interval integrability of functions with even or odd parity
+-/
+section
+
+variable {f : ℝ → E}
+
+/-- An even function is interval integrable (with respect to the volume measure)
+on every interval if it is interval integrable (with respect to the volume
+measure) on every interval of the form `0..x`, for positive `x`. -/
+
+theorem intervalIntegrable_of_even
+  (h₁f : ∀ x, f x = f (-x))
+  (h₂f : ∀ x, 0 < x → IntervalIntegrable f volume 0 x) :
+  ∀ x y, IntervalIntegrable f volume x y := by
+  -- Lemma: Prove statement first in case where x = 0
+  have : ∀ t, IntervalIntegrable f volume 0 t := by
```

Comment:
In general, formalisation experience says that extracting intermediate lemmas is almost always useful. Can you extract this `have` into its own lemma?